### PR TITLE
Ui 2457 bug visited route

### PR DESCRIPTION
**Pull request by sheela-svmx**
Scenario:

When there are multiple routes, if one route is a visited route and has a different color, on click on the visited route , then click on another route, the visited route defaulted to blue. 

This PR fixes that issue that was from our end. 

More details here 

https://servicemax.atlassian.net/browse/UI-2457

### CODE REVIEW

**Comment by sheela-svmx** [comment]
@mallik-svmx @joshsylvester @hanzge @dargonwanglong  When you get a chance review this .

**Comment by sheela-svmx** [comment]
I have a recording file I have uploaded on Box I have set the right viewing permissions https://ge.box.com/s/4v0wm7wh6ursy7sab419wotbfc4lfwwz , just how it is on my local.

**Comment by shao** [comment]
cc @leslieli

File: px-map-behavior-layer-geojson.es6.js [hunk 1]
```diff
@@ -342,16 +342,23 @@
...
 
     _handleFeatureTapped(evt) {
+      this.set('showFeatureProperties', 'false');
+
+      var features = this.data.features
+      var featureIdObj = {};
+      var uniqueFeatureArr = [];
+      for (var feature in features) {
+        if (!featureIdObj[features[feature].id]) {
+          featureIdObj[features[feature].id] = features[feature].id;
```
Comment (hanzge) [inline px-map-behavior-layer-geojson.es6.js]: This is how interview algorithms work in the real world.😄 Looks good to me.

**Pull request merged**